accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,233 @@
+              </li>
+            </ol>
+            {{addClass(listItem, "blocklyHasChildren")}}
+          </li>
+          <div *ngFor="#inputBlock of block.inputList">
+            <field-view *ngFor="#field of getInfo(inputBlock)" [field]="field"></field-view>
+            <tree-view *ngIf="inputBlock.connection && inputBlock.connection.targetBlock()" [block]="inputBlock.connection.targetBlock()" [isTopBlock]="false" [level]="level"></tree-view>
+            <li #inputList [attr.aria-level]="level+1" id="{{treeService.createId(inputList)}}" *ngIf="inputBlock.connection && !inputBlock.connection.targetBlock()" (keydown)="treeService.keyHandler($event, tree)">
+              {{getInputTypeLabel(inputBlock.connection)}} {{valueOrStatement(inputBlock)}} needed:
+              <select aria-label="insert input menu" (change)="inputMenuSelected(inputBlock.connection, $event)">
+                <option value="NO_ACTION" selected>select an action</option>
+                <option value="MARK_SPOT">mark this spot</option>
+                <option value="PASTE" disabled="{{clipboardCompatibilityHTMLText(inputBlock.connection)}}">paste</option>
+              </select>
+            </li>
+          </div>
+        </ol>
+      </li>
+      <tree-view *ngIf= "block.nextConnection && block.nextConnection.targetBlock()" [block]="block.nextConnection.targetBlock()" [isTopBlock]="false" [level]="level"></tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.TreeView; }), blocklyApp.FieldView],
+    inputs: ['block', 'isTopBlock', 'topBlockIndex', 'level', 'parentId'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService,
+      function(_clipboardService, _treeService) {
+      this.infoBlocks = {};
+      this.nextBlock = {};
+      this.sharedClipboardService = _clipboardService;
+      this.treeService = _treeService;
+    }],
+    setLabelledBy: function(item, string) {
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', string);
+      }
+    },
+    concatStringWithSpaces: function() {
+      var string = arguments[0];
+      for (i = 1; i < arguments.length; i++) {
+        string = string + ' ' + arguments[i];
+      }
+      return string;
+    },
+    noPreviousConnectionHTMLText: function(block) {
+      if (!block.previousConnection) {
+        return 'blockly-disabled';
+      } else {
+        return undefined;
+      }
+    },
+    noNextConnectionHTMLText: function(block) {
+      if (!block.nextConnection) {
+        return 'blockly-disabled';
+      } else {
+        return undefined;
+      }
+    },
+    checkParentList: function(parentList) {
+      blocklyApp.debug && console.log('setting parent list');
+      var tree = parentList;
+      while (tree && !tree.id.match(/^blockly-workspace-tree\d+$/)) {
```

Comment:
Performance tip (that probably does not mater here, but just for future reference): This parses and creates a new regex object every iteration and every call.  It is faster to set  a variable to `/^blockly-workspace-tree\d+$/` once then match against the variable each time.